packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelForm.tsx

```diff
@@ -171,13 +176,15 @@ export default function ColumnModelForm(props: ColumnModelFormProps) {
     return {}
   }, [validationErrors])
 
-  const showErrorOnMaxSizeField = !!errorsByField['maximumSize']
+  const showErrorOnMaxSizeField =
+    !!errorsByField['maximumSize'] && !isSmallerThanRecommendedSize
```

Comment:
Make sure that color changes from error to warning if isSmallerThanRecommendedSIze is true. This is possible because no cases causes the two to overlap.